packages/devtools_app/lib/src/performance/performance_model.dart

```diff
@@ -917,43 +917,26 @@ abstract class TimelineEvent extends TreeNode<TimelineEvent>
 class SyncTimelineEvent extends TimelineEvent {
   SyncTimelineEvent(TraceEventWrapper firstTraceEvent) : super(firstTraceEvent);
 
-  bool get isUiEventFlow => uiFrameNumber != null;
-
-  bool get isRasterEventFlow => rasterFrameNumber != null;
-
-  int get uiFrameNumber {
-    if (_uiFrameNumber != null) return _uiFrameNumber;
-    final animatorBeginFrameEvent = firstChildWithCondition(
-        (TimelineEvent event) => event.name.contains(uiEventName));
-    if (animatorBeginFrameEvent != null) {
-      try {
-        return int.parse(animatorBeginFrameEvent
-            .traceEvents.first.event.args[TraceEvent.frameNumberArg]);
-      } catch (_) {
-        return null;
-      }
-    }
-    return null;
+  int get frameNumberFromArgs {
+    final frameNumber = traceEvents.first.event.args[TraceEvent.frameNumberArg];
+    return frameNumber != null ? int.tryParse(frameNumber) : null;
   }
 
-  int _uiFrameNumber;
-
-  int get rasterFrameNumber {
-    if (_rasterFrameNumber != null) return _rasterFrameNumber;
-    final gpuRasterizerDrawEvent = firstChildWithCondition(
-        (TimelineEvent event) => event.name.contains(rasterEventName));
-    if (gpuRasterizerDrawEvent != null) {
-      try {
-        return int.parse(gpuRasterizerDrawEvent
-            .traceEvents.first.event.args[TraceEvent.frameNumberArg]);
-      } catch (_) {
-        return null;
-      }
-    }
-    return null;
-  }
+  /// Whether this event is contains the flutter frame identifier for the UI
+  /// thread in its trace event args.
+  bool get isUiFrameIdentifier => name.contains(uiEventName) &&
```

Comment:
also consider caching these if they are called more than once.
Would it be possible to perform an exact match on the name rather than using contains?
Contains can false positive and is slow. If an exact match isn't posible, using startsWith or endsWith is faster and less false positive prone.

Reply:
cached and used ==